x-pack/filebeat/input/awss3/acker.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
Since `eventACKTracker.Add` and `eventACKTracker.ACK` are asynchronous it seem it would be possible that `eventACKTracker.Wait` will return before every `Add` is really called and ack'ed.
What do you think?
```
func TestEventACKHandlerAsync(t *testing.T) {
	ctx, cancel := context.WithCancel(context.Background())
	t.Cleanup(cancel)

	acker := newEventACKTracker(ctx)
	acker.Add(1)
	wg := new(sync.WaitGroup)

	wg.Add(1)
	go func() {
		acker.ACK()
		<-time.After(1 * time.Second)
		acker.Add(1)
		wg.Done()
	}()

	acker.Wait()
	wg.Wait()
	acker.ACK()
	assert.EqualValues(t, 0, acker.pendingACKs)
}
```